src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/QuantitySearchValue.cs

```diff
@@ -63,14 +64,21 @@ public static QuantitySearchValue Parse(string s)
                 throw new FormatException(Core.Resources.MoreThanTwoTokenSeparatorSpecified);
             }
 
-            decimal quantity = decimal.Parse(parts[0], NumberStyles.Number, CultureInfo.InvariantCulture);
-            string system = parts.Count > 1 ? parts[1] : string.Empty;
-            string code = parts.Count > 2 ? parts[2] : string.Empty;
-
-            return new QuantitySearchValue(
-                system.UnescapeSearchParameterValue(),
-                code.UnescapeSearchParameterValue(),
-                quantity);
+            try
+            {
+                decimal quantity = decimal.Parse(parts[0], NumberStyles.Number, CultureInfo.InvariantCulture);
+                string system = parts.Count > 1 ? parts[1] : string.Empty;
+                string code = parts.Count > 2 ? parts[2] : string.Empty;
+
+                return new QuantitySearchValue(
+                    system.UnescapeSearchParameterValue(),
+                    code.UnescapeSearchParameterValue(),
+                    quantity);
+            }
+            catch
```

Comment:
We should avoid catch-all blocks

Reply:
Yep, that makes sense. Using `TryParse` solved this issue. This catch-all block has been removed.